app/components/signin_modal/view.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { Modal, Button, Input } from 'react-bootstrap';
+import SessionActions from 'actions/session';
+import SigninActions from 'actions/signin';
+import SigninStore from 'stores/signin';
+
+export default class SigninModal extends React.Component {
+  state = SigninStore.getState()
+
+  constructor() {
+    super();
+
+    this.changeState = ::this.changeState;
+  }
+
+  componentDidMount() {
+    SigninStore.listen(this.changeState);
+  }
+
+  componentWillUnmount() {
+    SigninStore.unlisten(this.changeState);
+  }
```

Comment:
just in order to get rid of duplicating `componentWillUnmount` and  `componentDidMount`, we could try to use [connectToStores](https://github.com/altjs/utils/blob/master/src/connectToStores.js) ?